### PR TITLE
Sign Up Issue fix from home page and checkout page

### PR DESCRIPTION
**Pull request by DeepaliAgrwl**
Please verify the changes done for sign Up issue on Home page and checkout page

### CODE REVIEW

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/signin/clientlibs/js/signin.js [hunk 1]
```diff
@@ -1,10 +1,15 @@
...
+const findRedriectUrl = function(domId) {
 		let redirectURL = "";
 		const params = new URLSearchParams(window.location.search);
 		const keys = ['signin.html', 'signup.html'];
 		if(params.has('referer')) {
 			if(!keys.some(k => params.get('referer').includes(k) )) {
 				redirectURL = params.get('referer');
+                if(!hasCart && redirectURL.includes('checkout.html'))
+                {
+					redirectURL = "/content/hclecomm/us/en/home.html";
```
Comment (rishabh-behl) [inline ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/signin/clientlibs/js/signin.js]: @DeepaliAgrwl : The path should not be hard coded . Please author default redirect path in sign in component

**Review by rishabh-behl — COMMENTED** [review]
@DeepaliAgrwl : Pl incorporate single comment

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/signin/clientlibs/js/signin.js (hunk 1 above)
Comment (DeepaliAgrwl) [reply ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/signin/clientlibs/js/signin.js]: Hard coded URL has been replaced by the default url authored in the component.

File: core/pom.xml [hunk 2]
```diff
@@ -188,5 +188,11 @@ Bundle-DocURL:
             <groupId>uk.org.lidalia</groupId>
             <artifactId>slf4j-test</artifactId>
         </dependency>
+        <dependency>
```
Comment (rishabh-behl) [inline core/pom.xml]: @DeepaliAgrwl : why do we need this dependency ?

File: core/src/main/java/com/hcl/ecomm/core/services/impl/CartServiceImpl.java [hunk 3]
```diff
@@ -80,6 +80,7 @@ public JsonArray getCartItemsDetails(String cartId, String customerToken) {
             CloseableHttpResponse httpResponse = httpClient.execute(httpGet);
             if(httpResponse.getStatusLine().getStatusCode() == 200)
             {
+
```
Comment (rishabh-behl) [inline core/src/main/java/com/hcl/ecomm/core/services/impl/CartServiceImpl.java]: @DeepaliAgrwl : remove extra line

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/clientlibs/clientlib-site/js.txt [hunk 4]
```diff
@@ -1 +1,2 @@
-js/utility.js
\ No newline at end of file
+js/utility.js
+js/loader.js
```
Comment (rishabh-behl) [inline ui.apps/src/main/content/jcr_root/apps/hclecomm/clientlibs/clientlib-site/js.txt]: @DeepaliAgrwl : why do we have loader as part of this PR ?

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/clientlibs/clientlib-site/js/loader.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
+
```
Comment (rishabh-behl) [inline ui.apps/src/main/content/jcr_root/apps/hclecomm/clientlibs/clientlib-site/js/loader.js]: @DeepaliAgrwl : why do we have loader as part of this PR ?

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/clientlibs/clientlib-site/js/utility.js [hunk 6]
```diff
@@ -26,4 +26,23 @@ return true;
 } else {
 return false;
 }
-}
\ No newline at end of file
+}
```
Comment (rishabh-behl) [inline ui.apps/src/main/content/jcr_root/apps/hclecomm/clientlibs/clientlib-site/js/utility.js]: @DeepaliAgrwl : why do we have loader as part of this PR ?

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/signin/clientlibs/js/signin.js [hunk 7]
```diff
@@ -1,10 +1,15 @@
...
+const findRedriectUrl = function(domId) {
 		let redirectURL = "";
 		const params = new URLSearchParams(window.location.search);
 		const keys = ['signin.html', 'signup.html'];
 		if(params.has('referer')) {
 			if(!keys.some(k => params.get('referer').includes(k) )) {
 				redirectURL = params.get('referer');
+                if(!hasCart && redirectURL.includes('checkout.html'))
+                {
+					redirectURL = domId.getAttribute("data-default");
```
Comment (rishabh-behl) [inline ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/signin/clientlibs/js/signin.js:11]: @dee: where is data-default defined in component ?

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/components/structure/page/customheaderlibs.html [hunk 8]
```diff
@@ -9,3 +9,4 @@
 <sly data-sly-include="/apps/hclecomm/components/structure/page/favicons.html"></sly>
 <!--/* Include Context Hub */-->
 <!--<sly data-sly-resource="${'contexthub' @ resourceType='granite/contexthub/components/contexthub'}"/> -->
+<span class ="checkmode" checkmode = "${wcmmode.edit ? 'edit' : 'notEdit'}" />
```
Comment (rishabh-behl) [inline ui.apps/src/main/content/jcr_root/apps/hclecomm/components/structure/page/customheaderlibs.html]: @DeepaliAgrwl : why do we have loader code here ?

**Review by rishabh-behl — CHANGES_REQUESTED** [review]
@DeepaliAgrwl : pl check comments

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/signin/clientlibs/js/signin.js (hunk 7 above)
Comment (DeepaliAgrwl) [reply ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/signin/clientlibs/js/signin.js:11]: It is already present in SignIn Form html. Like this

<form method="POST" action="" id="login_form" name="login_form" enctype="multipart/form-data"
	 class="cmp-login-form" data-default="${properties.defaultPage}.html"

**Comment by rakeshkumar-verma** [comment]
LGTM

**Pull request merged**